src/cpp/utils/shared_memory/SharedDir.hpp

```diff
@@ -67,6 +73,37 @@ class SharedDir
         return path + "/" + filename;
     }
 
+    static std::string get_lock_path(
+            const std::string& filename)
+    {
+#ifdef __QNXNTO__
+        static const char defaultdir[] = "/var/lock";
+        struct stat buf;
+        // check directory status
+        if (stat(defaultdir, &buf) != 0)
+        {
+            // directory not found, create it
+            if (errno == ENOENT)
+            {
+                mkdir(defaultdir, 0777);
+            }
+            // if another error then throw exception
+            else
+            {
+                std::string err("get_file_path() ");
+                err = err + strerror(errno);
+                throw std::runtime_error(err);
+            }
+        }
+        else
+        {
+            // directory exists do nothing
+        }
```

Comment:
this function does not return anything for when ```__QNXNTO__``` is defined.
add the following line:

```
return (std::string(defaultdir) + "/" + filename) ;
```

Reply:
@asobhy-qnx Thanks for the suggestion! I missed that on my review 🤦 

I have added the necessary change on 061a563. Please check that it works.

Reply:
Works, thank you.